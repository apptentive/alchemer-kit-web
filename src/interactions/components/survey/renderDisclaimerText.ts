/**
 * Renders a given surveys submit button.
 *
 * @param {string} text - The disclaimer text to render
 * @returns {HTMLDivElement} - The DOM Node
 */
const renderDisclaimerText = (text: string) => {
  const disclaimerText = document.createElement('p');
  disclaimerText.classList.add('apptentive-disclaimer__text');
  disclaimerText.classList.add('apptentive-typography__description');
  disclaimerText.classList.add('apptentive-typography__description--tall');
  disclaimerText.textContent = text;

  // If the disclaimer text is long, decrease the font size slightly to provide more room for the rest of the interaction
  if (text.length > 160) {
    disclaimerText.style.fontSize = '12px';
  }

  const disclaimerWrapper = document.createElement('div');
  disclaimerWrapper.classList.add('apptentive-disclaimer');
  disclaimerWrapper.appendChild(disclaimerText);

  return disclaimerWrapper;
};

export default renderDisclaimerText;
