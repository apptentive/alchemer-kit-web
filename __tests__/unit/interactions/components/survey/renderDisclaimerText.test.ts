import renderDisclaimerText from '../../../../../src/interactions/components/survey/renderDisclaimerText';

const _shortDisclaimerText = 'Short Disclaimer';
const _longDisclaimerText =
  'Long disclaimer text that is more than 160 characters and really should not be this long but sometimes it is unavoidable so we have to handle it gracefully and decrease the font size';

describe('renderDisclaimerText', () => {
  test('properly renders short disclaimer text', () => {
    const output = renderDisclaimerText(_shortDisclaimerText);

    expect(output).toMatchSnapshot();
  });

  test('properly renders long disclaimer text', () => {
    const output = renderDisclaimerText(_longDisclaimerText);

    expect(output).toMatchSnapshot();
  });
});
