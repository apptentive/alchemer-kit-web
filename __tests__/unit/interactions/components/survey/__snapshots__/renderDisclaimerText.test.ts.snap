// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderDisclaimerText properly renders long disclaimer text 1`] = `
<div
  class="apptentive-disclaimer"
>
  <p
    class="apptentive-disclaimer__text apptentive-typography__description apptentive-typography__description--tall"
    style="font-size: 12px;"
  >
    Long disclaimer text that is more than 160 characters and really should not be this long but sometimes it is unavoidable so we have to handle it gracefully and decrease the font size
  </p>
</div>
`;

exports[`renderDisclaimerText properly renders short disclaimer text 1`] = `
<div
  class="apptentive-disclaimer"
>
  <p
    class="apptentive-disclaimer__text apptentive-typography__description apptentive-typography__description--tall"
  >
    Short Disclaimer
  </p>
</div>
`;
